src/structify/resources/structure.py

```diff
@@ -277,17 +277,17 @@ def run(  # type: ignore
         start_time = time.time() if timeout is not None else None
 
         while True:
-            status = self.job_status(body=[token])
-
-            if status.body["status"] == "completed":  # type: ignore
-                return self._client.datasets.view(dataset_name=kwargs["dataset_name"], table_name=table_name) # type: ignore
-
             if timeout is not None and start_time is not None:
                 elapsed_time = time.time() - start_time
                 if elapsed_time > timeout:
-                    # TODO: Cancel hasn't been merged yet.
-                    # self._client.runs.cancel(token)
-                    raise TimeoutError(f"Job exceeded timeout of {timeout} seconds and was cancelled.")
+                    raise TimeoutError(f"Job creation or execution exceeded timeout of {timeout} seconds.")
```

Comment:
It might be helpful for the users to know which one - you can tell by whether the try statement has been successful and set a flag